### PR TITLE
Added searching in private repos & added sorting to organizations, repos & users page

### LINKED ISSUE #207: Permissions related question
Hi there, 

FYI : I have added the same question on Gogs official repo as both projects are linked. 

Not really a bug but a question about the permissions : they are not very clear to me, am I the only one?

I want to use Gogs as a **private repositories hosting** app. When installing Gogs, I have set in the options that Gogs is private, no registration, no public page. This part works. 

I have created an organisation named **MyOrg** and I have created 10 repositories whose owner is **MyOrg** (not me). When visiting the "Explorer" page as me (admin), I can list my 10 repos. 
Then I have created a team **MyTeam** in **MyOrg** with write permission in which I have added an user "**user1**". And I have created another user "**user2**" with no team.

When logging as "**user2**", this user could see the 10 repos in "Explorer". I was at first surprised because this user does not belong to any organisation nor team then I realized that I had to set "Private" in the repo settings. Doing so, "**user2**" could not see any repo any longer, which was good. 

When logging as "**user1**", this user could not see any repo in "Explorer" despite it belongs to the team in the organisation. Also, even when browsing on the organisation page, it could not see any repo. 
Once again, I realized I had to manually add each repo to the team and this is weird and time consuming. Doing so, the "**user1**" can now see the repos from the organisation page (but not from the "Explorer" page, which is weird). So I got to the point where my 10 repos are owned by me (Owner) and as a duplicate by the team (MyTeam). Plus adding a repo to a team is not very user-friendly as you must remember the name of the repo for autocompletion (so I did this in the database directly). 

Finally, I was really surprised when I realized that me (Owner and admin) could not see any repo in Explorer any longer. I had to go through the organisation page. 

Taking Bitbucket as an example, it is not the way Bitbucket works. Once a team is affected to an organisation, the team has permission to list and/or write (according to the given permission) on all repos of this organisation. It is not required to manually affect each repo to the team as these repos are already affected to the organisation. Plus, on Bitbucket, when listing the repos, it will output all the repos you have access to. I don't understand why the "Explorer" page is now empty. 

Is there anything I missed? 
Sorry for the long post, wasn't easy to be shorter. 
Thanks

Axel

### PR DESCRIPTION
**Pull request by Bwko**
Added searching in private repos
Added sorting to organizations, repos & users page
Fixes #207
Needs #199 for translations

### CODE REVIEW

File: models/org.go [hunk 1]
```diff
@@ -189,12 +189,19 @@ func CountOrganizations() int64 {
...
-func Organizations(page, pageSize int) ([]*User, error) {
-	orgs := make([]*User, 0, pageSize)
-	return orgs, x.
-		Limit(pageSize, (page-1)*pageSize).
-		Where("type=1").
-		Asc("name").
+func Organizations(opts *SearchUserOptions) ([]*User, error) {
+	orgs := make([]*User, 0, opts.PageSize)
+
+	if len(opts.OrderBy) == 0 {
```
Comment (strk) [inline models/org.go:201]: I keep encountering these `len(val) == 0` constructs, why not `val == ""` instead ?

**Comment by strk** [comment]
LGTM (only read the diff, not tested)

File: models/org.go (hunk 1 above)
Comment (Bwko) [reply models/org.go:201]: @strk it is a micro optimization http://stackoverflow.com/a/18595217

**Comment by strk** [comment]
The suggestion from Russ Cox is "make the code clear".
Personally, I find

  `mystring != ""`

clearer than

  `len(mystring) > 0`

**Comment by Bwko** [comment]
Both options are clear enough to me. But since the second option is marginally faster(and used the most  in gitea) I prefer the second notation.

**Comment by ghost** [comment]
LGTM

**Comment by axeloz** [comment]
Hello, 

any idea when this will be merged or released? I'm still a bit stucked without this. 
Thanks

File: models/org.go (hunk 1 above)
Comment (tboerger) [reply models/org.go:201]: I've got to say that I also prefer the `if opts.OrderBy == ""` variant.

**Comment by tboerger** [comment]
> Hello,
> 
> any idea when this will be merged or released? I'm still a bit stucked without this.
> Thanks

This will be part of 1.1.0 because we are already in a hurry to get the first release done.

**Comment by lunny** [comment]
Please resolve the conflict and we will merge it.

**Comment by lunny** [comment]
LGTM work

**Pull request merged**